x-pack/plugins/enterprise_search/server/lib/indices/delete_ml_inference_pipeline.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IngestPutPipelineRequest } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '@kbn/core/server';
+
+/**
+ * Response for deleting sub-pipeline from @ml-inference pipeline.
+ * If sub-pipeline was deleted successfully, 'deleted' field contains its name.
+ * If parent pipeline was updated successfully, 'updated' field contains its name.
+ */
+export interface DeleteMlInferencePipelineResponse {
+  deleted?: string;
+  updated?: string;
+}
+
+export const deleteMlInferencePipeline = async (
+  indexName: string,
+  pipelineName: string,
+  client: ElasticsearchClient
+) => {
+  const response: DeleteMlInferencePipelineResponse = {};
+  const parentPipelineId = `${indexName}@ml-inference`;
+
+  // find parent pipeline
+  try {
+    const pipelineResponse = await client.ingest.getPipeline({
+      id: parentPipelineId,
+    });
+
+    const parentPipeline = pipelineResponse[parentPipelineId];
+
+    if (parentPipeline !== undefined) {
+      // remove sub-pipeline from parent pipeline
+      if (parentPipeline.processors !== undefined) {
+        const updatedProcessors = parentPipeline.processors.filter(
+          (p) => !(p.pipeline !== undefined && p.pipeline.name === pipelineName)
+        );
+        // only update if we changed something
+        if (updatedProcessors.length !== parentPipeline.processors.length) {
+          const updatedPipeline = {
+            ...parentPipeline,
+            ...{ processors: updatedProcessors },
+          } as IngestPutPipelineRequest;
+
+          const updateResponse = await client.ingest.putPipeline(updatedPipeline);
+          if (updateResponse.acknowledged === true) {
+            response.updated = parentPipelineId;
+          }
+        }
+      }
+    }
+  } catch (error) {
+    // only catch Not Found error
+    if (error.meta?.statusCode !== 404) {
+      throw error;
+    }
+  }
+
+  // finally, delete pipeline
+  const deleteResponse = await client.ingest.deletePipeline({ id: pipelineName });
```

Comment:
What happens if this throws an error?

Reply:
Let it. The router handles errors.

Reply:
if the pipelineName doesn't exist, does it throw an exception? If it does, are we happy the eventual response will be a 502?

Reply:
Yes, it throws an exception. Elasticsearch will return a 404.

I have no idea why we're converting all exception (except "unauthorized") into 502 error codes. I assumed it must make sense on the frontend, because that's what all the other routes do. Better ask someone with more frontend knowledge, maybe @byronhulcher.

Reply:
typically we would throw a pipeline doesn't exist errorcode here and catch the errorcode on the router. The router will call `createError` with a more descriptive message that the pipeline couldn't be found. You do a similar pattern for index not found in your other PR.

On the frontend, we have a number of options. Typically we will display the API error message.

Reply:
Message has to be localized, right? How do you do that? And where can I see what localized messages we already have?

Reply:
So, I cheated here, and returned an error with message I have from Elasticsearch (`error.meta?.body?.error?.reason`), this way I don't have to localize it. I saw the documentation:
https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/GUIDELINE.md#development-steps
I just don't fully comprehend all the steps.

Reply:
I think thats fine for now. I suggest we revisit it under tech debt for localisation once the frontend build out settles and have more understanding on how the error display message will appear.